### PR TITLE
Dmitrii/refactor

### PR DESCRIPTION
**Pull request by dgubko**
# Description

Final refactor and styles

# Fixes # (issue)

1. Added missing types
2. Added styles.
3. #55 Fixed bug clearing search bar after search
4. Deleted redundant Promise.resolve
5. Deleted redundant event argument on handleChange
6. Removed unused code.

# Type of change

- [x]  Bug fix (non-breaking change which fixes an issue)
 
- [ ]  New feature (non-breaking change which adds functionality)

- [x]  Breaking change (fix or feature that would cause existing functionality not to work as expected)

- [ ] This change requires a documentation update

-  [x] Styling

- [x] Refactor

- [x] Testing

# Where were the changes?

All over.

# Checklist:

- [x]  My code follows the style guidelines of this project
- [x]  I have performed a self-review of my code
- [ ]  I have commented my code, particularly in hard-to-understand areas (if applicable)
- [ ]  I have made corresponding changes to the documentation (if applicable)
- [x]  My changes generate no new warnings

# Contributors (paired programming):
@Nagel29 @Adelle-Pitsas

### CODE REVIEW

**Review by Courtney-Lippman — APPROVED** [review]
Looks good great refactors!

File: cypress/e2e/details.cy.js [hunk 1]
```diff
@@ -19,58 +22,67 @@ describe("Game details when API server is running", () => {
   });
 
   it("should display a title", () => {
-    cy.get("#title").contains("Boardgame");
+    cy.get("#title").contains("Board Game");
```
Comment (knpausch) [inline cypress/e2e/details.cy.js:25]: Thanks for the grammatical correction here!

File: src/components/App/App.css [hunk 2]
```diff
@@ -12,6 +13,12 @@ header {
   border-radius: 8px;
   margin-bottom: 16px;
   justify-content: space-between;
+  font-family: "Permanent Marker", cursive;
```
Comment (knpausch) [inline src/components/App/App.css:16]: Cool styling!

**Review by knpausch — APPROVED** [review]
Thanks for tackling the refactor and DRYing up the code!

**Pull request merged**